subsys/bluetooth/services/fast_pair/fp_storage/include/fp_storage_ak_bond.h

```diff
@@ -0,0 +1,143 @@
+
+/** Indicate that the bonding process has been finalized.
+ *
+ *  It relates to the Account Key association with the Bluetooth bond that was created
+ *  earlier using the @ref fp_storage_ak_bond_conn_create API. This function should be called when
+ *  the BLE conn disconnects to let the storage subsystem handle incomplete bonds.
+ *
+ *  If the @ref fp_storage_ak_bond_conn_create API has not been called for this conn earlier,
+ *  the call is ignored.
+ *
+ * @param[in] conn_ctx Connection context associated with the Bluetooth bond.
+ */
+void fp_storage_ak_bond_conn_finalize(const void *conn_ctx);
+
+/** Update the address of the Bluetooth bond associated with the Account Key.
+ *
+ *  If the @ref fp_storage_ak_bond_conn_create API has not been called for this conn earlier,
+ *  the call is ignored.
+ *
+ * @param[in] conn_ctx Connection context associated with the Bluetooth bond.
+ * @param[in] addr The Bluetooth address of the bond.
+ */
+void fp_storage_ak_bond_conn_addr_update(const void *conn_ctx, const bt_addr_le_t *new_addr);
+
+/** Delete the Account Key association with the Bluetooth bond.
+ *
+ *  If the @ref fp_storage_ak_bond_conn_create API has not been called for this conn earlier,
+ *  the call is ignored.
+ *
+ * @param[in] addr The Bluetooth address of the bond.
+ */
+void fp_storage_ak_bond_delete(const bt_addr_le_t *addr);
+
+/** Bluetooth bond request callback structure. */
+struct fp_storage_ak_bond_bt_request_cb {
+	/** @brief Remove the Bluetooth bond.
+	 *
+	 *  This callback is called to request the removal of the Bluetooth bond from the Bluetooth
+	 *  subsystem.
+	 *
+	 * @param[in] addr The Bluetooth address of the bond to be removed.
+	 *
+	 * @return 0 If the operation was successful. Otherwise, a (negative) error code is
+	 *	   returned.
+	 */
+	int (*bond_remove)(const bt_addr_le_t *addr);
+
+	/** @brief Check if the Bluetooth address is bonded.
+	 *
+	 *  This callback is called to check if there is a Bluetooth bond saved in the Bluetooth
+	 *  subsystem with the given Bluetooth address.
+	 *
+	 * @param[in] addr The Bluetooth address.
+	 *
+	 * @return true when the Bluetooth address is bonded, false otherwise.
+	 */
+	bool (*is_addr_bonded)(const bt_addr_le_t *addr);
+};
+
+/** Register Bluetooth bond request callbacks.
+ *
+ *  This function registers an instance of request callbacks. The registered instance needs to
+ *  persist in the memory after this function exits, as it is used directly without the copy
+ *  operation. It is possible to register only one instance of callbacks with this API.
+ *
+ *  This function must be called before initialiazing the Fast Pair storage with the
+ *  @ref fp_storage_init API if the CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND Kconfig option is enabled.
+ *  This function must be called before performing the Fast Pair factory reset if the
+ *  CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND Kconfig option is enabled.
```

Comment:
`This function should be registered early to ensure that the callbacks are available both during Fast Pair factory reset and on Fast Pair storage initialization`? I think there is no need to mention `CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND` as whole header is used only if `CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND` is enabled

Reply:
Changed to
```
 *  This function must be called before initialiazing the Fast Pair storage with the
 *  @ref fp_storage_init API and also before performing the Fast Pair factory reset to ensure that
 *  the callbacks are available during those operations.
```